spark-cassandra-connector-java/src/main/java/com/datastax/spark/connector/japi/rdd/CassandraJavaPairRDDLike.java

```diff
@@ -0,0 +1,31 @@
+package com.datastax.spark.connector.japi.rdd;
+
+import com.datastax.spark.connector.cql.CassandraConnector;
+import com.datastax.spark.connector.NamedColumnRef;
+import com.datastax.spark.connector.rdd.CassandraRDD;
+import com.datastax.spark.connector.rdd.ReadConf;
+import scala.Tuple2;
+import scala.reflect.ClassTag;
+
+import static com.datastax.spark.connector.util.JavaApiHelper.toScalaSeq;
+
+/**
+ * A Java API wrapper over {@link com.datastax.spark.connector.rdd.CassandraRDD} of tuples to provide Spark Cassandra
+ * Connector functionality in Java.
+ *
+ * <p>The wrapper can be obtained by one of the methods of {@link com.datastax.spark.connector.japi.SparkContextJavaFunctions}
+ * or {@link com.datastax.spark.connector.japi.StreamingContextJavaFunctions}.</p>
+ */
+public interface CassandraJavaPairRDDLike<K, V, ThisType extends CassandraJavaPairRDDLike<K, V, ThisType>> {
```

Comment:
I would change the name to just CassandraJavaPairRDD because this type will be used by the users. For the implementation, just add Impl suffix.